### PR TITLE
RfP for 2020 Q1 for pip

### PR DESCRIPTION
**Pull request by brainwane**
Work in progress; please critique @ewdurbin and @pradyunsg.

### CODE REVIEW

File: 2020-pip/RFP.md [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+
+Familiarity and expertise with all technologies is not required. Strong Python skills and experience are a must, though.
+
+#### Specific Technologies Used
+
+For the best primer, see the [developer documentation](https://pip.pypa.io) for pip.
+
+You can also see the [complete codebase](https://github.com/pypa/pip) on GitHub.
+
+????? test infrastructure
```
Comment (ewdurbin) [inline 2020-pip/RFP.md]:
```markdown
##### Testing Infrastructure

pip has an extensive continuous integration test suite. This test suite is built using

- [tox](https://tox.readthedocs.io/en/latest/)
- [nox](https://nox.thea.codes/en/stable/)
- [pre-commit](https://pre-commit.com)
- [pytest](http://doc.pytest.org/en/latest/)

As well as many other associated Python testing, automation, and code quality tools.

Components of the test suite are executed using multiple services including:

- [Azure Pipelines](https://docs.microsoft.com/en-us/azure/devops/pipelines/index?view=azure-devops)
- [GitHub Actions](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-github-actions)
- [AppVeyor](https://www.appveyor.com/docs/)
- [TravisCI](https://docs.travis-ci.com)
- [Read the Docs](https://docs.readthedocs.io/en/stable/index.html)

```

File: 2020-pip/RFP.md [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+
+**A Request for Proposals (RFP)** is a process intended to allow us (the Python Software Foundation) to collect proposals from potential contractors and select contractor(s) best suited to fulfill the specified work.
+
+After the RFP period closes we will evaluate the received proposals based on the [evaluation criteria](#evaluation-criteria), seek clarification from proposers as necessary, and select one or more contractors to complete the work specified in the [scope section](#scope).
+
+**Note**: This Request For Proposal document may be updated to reflect things that we learn during the process. The canonical version and history is available [here](https://github.com/python/request-for/blob/master/2020-pip/RFP.md).
+
+## How do I submit a proposal?
+
+Proposals should be submitted as [Portable Document Format (PDF)](https://en.wikipedia.org/wiki/PDF) files via email to [ewdurbin@pyfound.org](mailto:ewdurbin@pyfound.org).
```
Comment (ewdurbin) [inline 2020-pip/RFP.md]: Please include a Subject line prefix for identifying proposals.

File: 2020-pip/RFP.md [hunk 3]
```diff
@@ -0,0 +1,273 @@
...
+##### Backend
+
+While pip is a client-side program, there will be backend work involved in building test infrastructure.
+
+Travis CI
+Azure?
+We're open
+
+- [Python](https://www.python.org/) 3.6+
+- [PostgreSQL](https://www.postgresql.org) - [SQLAlchemy](https://docs.sqlalchemy.org/en/latest/) - [psycopg2](http://initd.org/psycopg/)
```
[the remark this answers is not in the document]
Comment (brainwane) [reply 2020-pip/RFP.md]: Thanks. Fixed.

File: 2020-pip/RFP.md [hunk 4]
```diff
@@ -0,0 +1,273 @@
...
+- [Redis](https://redis.io)
+
+##### Frontend
+
+Currently the only frontend work for pip is the pip documentation, which is built using reStructuredText and Sphinx. While building test infrastructure, we may want to add some frontend capability to that. The Python Package Index uses the following, so we have some team expertise in them and would prefer that those be used for any frontend work.
+
+Note: Our frontend is primarily static; these tools power the toolchain that creates our final assets.
+
+- [Node](https://nodejs.org/en/)
```
[the remark this answers is not in the document]
Comment (brainwane) [reply 2020-pip/RFP.md]: Thanks. Removed.

File: 2020-pip/RFP.md [hunk 5]
```diff
@@ -0,0 +1,273 @@
...
+
+##### Backend
+
+While pip is a client-side program, there will be backend work involved in building test infrastructure.
+
+Travis CI
+Azure?
+We're open
+
+- [Python](https://www.python.org/) 3.6+
```
[the remark this answers is not in the document]
Comment (brainwane) [reply 2020-pip/RFP.md]: Yes. Fixed.

File: 2020-pip/RFP.md (hunk 2 above)
Comment (brainwane) [reply 2020-pip/RFP.md]: Thanks. Added.

File: 2020-pip/RFP.md (hunk 1 above)
Comment (brainwane) [reply 2020-pip/RFP.md]: Thanks. Added.

**Comment by brainwane** [comment]
Ready for re-review.

File: 2020-pip/RFP.md [hunk 6]
```diff
@@ -0,0 +1,278 @@
...
+
+This project is intended to be completed over three phases, totalling a little over one year, starting mid-December 2019 or early January 2020.
+
+Role 1 and Role 2 have different timelines; see those roles for details.
+
+### Technology
+
+The [pip codebase](https://github.com/pypa/pip) is licensed under the [MIT License](https://github.com/pypa/pip/blob/master/LICENSE.txt). All work submitted or dependencies added must be compliant with this license.
+
+The backend codebase is in Python. The pip documentation is built using reStructuredText and Sphinx.
```
Comment (pradyunsg) [inline 2020-pip/RFP.md]: Which backend does this refer to?

Comment (brainwane) [reply 2020-pip/RFP.md]: Ah right; sorry, "backend" here is extraneous and I will remove it.

**Comment by brainwane** [comment]
@ewdurbin If you want, I can receive the applications and questions at my pypi.org address to save you some initial screening work.

File: 2020-pip/RFP.md [hunk 7]
```diff
@@ -0,0 +1,278 @@
...
+You can also see the [complete codebase](https://github.com/pypa/pip) on GitHub.
+
+ - [Python](https://www.python.org/) 2.7 and 3.5+ on Linux, Mac OS, and Windows
+ - [pytest](http://pytest.readthedocs.io/)
+ - [tox](https://tox.readthedocs.io/en/latest/)
+ - [Sphinx](http://www.sphinx-doc.org/)
+
+##### Backend and Testing Infrastructure
+
+pip has an extensive continuous integration test suite. This test suite is built using:
```
Comment (pradyunsg) [inline 2020-pip/RFP.md]: We have a WIP document describing this too -- https://pip.pypa.io/en/latest/development/ci/. It may be worth linking to it.

**Comment by ewdurbin** [comment]
> @ewdurbin If you want, I can receive the applications and questions at my pypi.org address to save you some initial screening work.

That sounds great to me.

File: 2020-pip/RFP.md [hunk 8]
```diff
@@ -0,0 +1,278 @@
...
+For the best primer, see the [developer documentation](https://pip.pypa.io) for pip.
+
+You can also see the [complete codebase](https://github.com/pypa/pip) on GitHub.
+
+ - [Python](https://www.python.org/) 2.7 and 3.5+ on Linux, Mac OS, and Windows
+ - [pytest](http://pytest.readthedocs.io/)
+ - [tox](https://tox.readthedocs.io/en/latest/)
+ - [Sphinx](http://www.sphinx-doc.org/)
+
+##### Backend and Testing Infrastructure
```
Comment (pradyunsg) [inline 2020-pip/RFP.md]: I guess this "Backend" is also extraneous?

```suggestion
##### Testing Infrastructure
```

File: 2020-pip/RFP.md [hunk 9]
```diff
@@ -0,0 +1,278 @@
...
+- [GitHub Actions](https://help.github.com/en/actions/automating-your-workflow-with-github-actions/about-github-actions)
+- [AppVeyor](https://www.appveyor.com/docs/)
+- [TravisCI](https://docs.travis-ci.com)
+- [Read the Docs](https://docs.readthedocs.io/en/stable/index.html)
+
+### Project Management and Reporting
+
+This project will be led and managed by the Python Software Foundation Director of Infrastructure and an external project manager.
+
+Regular meetings (at most 30 minutes each, 4 times per month) will be held to coordinate efforts among the project managers, backend developers, and UX researcher/designer.
```
Comment (pradyunsg) [inline 2020-pip/RFP.md]: I guess this "backend" is also extraneous?

```suggestion
Regular meetings (at most 30 minutes each, 4 times per month) will be held to coordinate efforts among the project managers, developers, and UX researcher/designer.
```

**Comment by pradyunsg** [comment]
LGTM, other than the few comments I've made. :)

File: 2020-pip/RFP.md (hunk 7 above)
Comment (brainwane) [reply 2020-pip/RFP.md]: Linked. Thanks.

File: 2020-pip/RFP.md (hunk 8 above)
Comment (brainwane) [reply 2020-pip/RFP.md]: Yes, thanks, fixed.

File: 2020-pip/RFP.md (hunk 9 above)
Comment (brainwane) [reply 2020-pip/RFP.md]: Yes, thanks, fixed.

**Comment by brainwane** [comment]
@ewdurbin I believe this is fine to merge now.

**Pull request merged**